### PR TITLE
Integrate new LavaMoat scuttling protection feature

### PR DESCRIPTION
**Pull request by weizman**
> Continue work from #15581 after messing it up

## Motivation

Integrating into MetaMask browser extension the new scuttling feature introduced by LavaMoat ([#360](https://github.com/LavaMoat/LavaMoat/pull/360))

## PR Breakdown

1. lockdown more collides with scuttling feature, adjust it to skip scuttled props
2. `scuttleGlobalThisExceptions` conf prop is populated by 3 different groups: (1) globals that are used by different mm deps that unsafely access global object props and therefore mess with scuttling; (2) globals the chrome driver uses in the test env of mm which cannot be sandboxed and therefore must be excused; and (3) globals vetted shim sentry needs to write to global object in order to function. **All of these will be narrowed down with time and research in different PRs**
3. Also, in order for sentry to work as the vetted shim that it is, we need to allow it to still access scuttled properties, therefore we wrap it with a special sandbox (a mix of `with`s and `proxy`s) that allows it to function normally in a scuttled envirnoment (partially by indicating a list of the properties it'll need that will be later scuttled)
4. update `test/helpers/protect-intrinsics-helpers.js` test to test intrinsics protection only for non scuttled props

In this PR currently there are patch-package files, only to apply the needed upgrade for LM - those will be removed before merge and will be replaced with an actual newly delivered version of LM

## Notes

This is an ongoing effort which will bring with it more PRs in the future. In following PRs we should focus on narrowing down as much as possible the scuttling avoids list which currently contains more fields we skip scuttling for then we'd want to ideally, some of them are potentially quite useful for attackers.

This will require some research.

### CODE REVIEW

**Comment by weizman** [comment]
Progress and comments on this PR up until this comment can be found in the previous PR #15581 that was closed and moved here after I messed it up by accident

File: .yarn/patches/@lavamoat-lavapack-npm-3.3.0-5a61b5374d.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/runtime.js b/src/runtime.js
+index 97c9ddb9b76a2fe31cf9b202cb491f15232232ff..26f65a2b2a3578638402fa887156fdad18eee368 100644
+--- a/src/runtime.js
++++ b/src/runtime.js
+@@ -8266,7 +8266,7 @@ function                tameFunctionConstructors() {
+     FERAL_FUNCTION.prototype.constructor('return 1');
+    }catch( ignore) {
+     // Throws, no need to patch.
+-    return harden({});
```
Comment (weizman) [inline .yarn/patches/@lavamoat-lavapack-npm-3.3.0-5a61b5374d.patch:9]: until https://github.com/endojs/endo/issues/1281 is fixed

File: .yarn/patches/lavamoat-core-npm-12.4.0-cecca1a9b5.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/lib/lockdown.umd.js b/lib/lockdown.umd.js
+index 2f32ec5014adc6006ad5148a759fbf48d46bb40c..6df14835df7ca594a775c7da639de7b657e229d7 100644
+--- a/lib/lockdown.umd.js
++++ b/lib/lockdown.umd.js
+@@ -8154,7 +8154,7 @@ function                tameFunctionConstructors() {
+     FERAL_FUNCTION.prototype.constructor('return 1');
+    }catch( ignore) {
+     // Throws, no need to patch.
+-    return harden({});
```
Comment (weizman) [inline .yarn/patches/lavamoat-core-npm-12.4.0-cecca1a9b5.patch:9]: until https://github.com/endojs/endo/issues/1281 is fixed

File: .yarn/patches/@lavamoat-lavapack-npm-3.3.0-5a61b5374d.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/runtime.js b/src/runtime.js
```
Comment (weizman) [inline .yarn/patches/@lavamoat-lavapack-npm-3.3.0-5a61b5374d.patch:1]: until https://github.com/endojs/endo/issues/1281 is fixed

File: .yarn/patches/lavamoat-core-npm-12.4.0-cecca1a9b5.patch [hunk 4]
```diff
@@ -0,0 +1,13 @@
+diff --git a/lib/lockdown.umd.js b/lib/lockdown.umd.js
```
Comment (weizman) [inline .yarn/patches/lavamoat-core-npm-12.4.0-cecca1a9b5.patch:1]: until https://github.com/endojs/endo/issues/1281 is fixed

File: .yarn/patches/async-done-npm-1.3.2-1f0a4a8997.patch [hunk 5]
```diff
@@ -0,0 +1,43 @@
+diff --git a/index.js b/index.js
```
Comment (weizman) [inline .yarn/patches/async-done-npm-1.3.2-1f0a4a8997.patch:1]: [Endo-SES](https://github.com/endojs/endo/) new version introduces "domain taming" which prevents systems from using the node builtin "domain" package due to security reasons.
Here we remove the usage "async-done" package is making of "domain" package so MM build system continues to work properly.

**Review by brad-decker — APPROVED** [review]
LGTM :)

**Review by kumavis — APPROVED** [review]

**Pull request merged**